source/main/utils/handleCheckBlockReplayProgress.ts

```diff
@@ -43,35 +44,57 @@ function getProgressType(line: string): GetBlockSyncProgressType | null {
 
 const applicationStartDate = moment.utc();
 
-export const handleCheckBlockReplayProgress = (
-  mainWindow: BrowserWindow,
-  logsDirectoryPath: string
-) => {
-  const filename = 'node.log';
-  const logFilePath = `${logsDirectoryPath}/pub/`;
-  const filePath = path.join(logFilePath, filename);
-  if (!fs.existsSync(filePath)) return;
-
-  const tail = new Tail(filePath);
+const createHandleNewLogLine = (mainWindow: BrowserWindow) => {
+  const lastReportedProgressByType: Record<BlockSyncType, number> = {
+    [BlockSyncType.pushingLedger]: 0,
+    [BlockSyncType.replayedBlock]: 0,
+    [BlockSyncType.validatingChunk]: 0,
+  };
 
-  tail.on('line', (line) => {
+  return (line: string) => {
     if (
       !isItFreshLog(applicationStartDate, line) ||
       !containProgressKeywords(line)
     ) {
       return;
     }
 
-    const percentage = line.match(/Progress:([\s\d.,]+)%/)?.[1];
-    const progressType = getProgressType(line);
-    if (!percentage || !progressType) {
+    const unparsedProgress = line.match(/Progress:([\s\d.,]+)%/)?.[1];
+    const type = getProgressType(line);
+    if (!unparsedProgress || !type) {
       return;
     }
-    const finalProgressPercentage = parseFloat(percentage);
-    // Send result to renderer process (NetworkStatusStore)
-    getBlockSyncProgressChannel.send(
-      { progress: finalProgressPercentage, type: progressType },
-      mainWindow.webContents
-    );
+
+    const progress = Math.floor(parseFloat(unparsedProgress));
+
+    if (lastReportedProgressByType[type] !== progress) {
```

Comment:
This helps us ensure we're not sending duplicate events via IPC, and mitigates the risk of seeing the `MaxListenersExceededWarning` which was previously seen while trying to send multiple (10+) messages at once. That could occur with `fs.fileWatch`, which is pooling files every N seconds and invoking given callback for each line (so if there are 100 new lines added in N seconds = 100 callback calls).